### DISCUSSION TITLE
How to use graphql-request's batching via the batchRequests() function in redux-saga?

### DISCUSSION THREAD

**Discussion by rishavbharti** [prisma/prisma#12791] [Q&A]
I'm trying to implement batching in redux-saga using graphql-request's `batchRequests()` function but not sure of the syntax.

Here's what I've tried

        export default ({ gqlClient } = {}) => {
    	    function* getDashboardData() {
    		while (true) {
    			try {
    				yield take(dashboardActions.getDashboardDataPending.type);
    
                    yield call([gqlClient, gqlClient.batchRequests],
    				   { document: queries.home.getUserData, variables: { period: 'day', date: 'last30' } },
    				   { document: queries.home.getRevenueData, variables: { period: 'day', date: 'last60' } }
    				 );
    				 
    				 ...
    		        }
                }
            }
        }

```
   export default ({ gqlClient } = {}) => {
    	    function* getDashboardData() {
    		while (true) {
    			try {
    				yield take(dashboardActions.getDashboardDataPending.type);
    
                    yield apply(gqlClient, gqlClient.batchRequests,
    				 [
    				   { document: queries.home.getUserData, variables: { period: 'day', date: 'last30' } },
    				   { document: queries.home.getRevenueData, variables: { period: 'day', date: 'last60' } }
    				 ]);
    				 
    				 ...
    		        }
                }
            }
        }
```

But this doesn't work.
[Error screenshot][1]
Using graphql-request v3.7.0 & redux-saga v1.1.3

Can someone tell the correct way to do it?
[graphql-request's docs][2]

  [1]: https://i.stack.imgur.com/09bEQ.png
  [2]: https://github.com/prisma-labs/graphql-request#batching

**Comment by nurul3101** [comment]
Hey @rishavbharti 👋 

In your `yield call` function argument, can you try passing the document objects in the form of an array?
So second and third arguments should be combined into an array instead.

Like this:
```javascript
yield call([gqlClient, gqlClient.batchRequests],
	[ { document: queries.home.getUserData, variables: { period: 'day', date: 'last30' } },
	 { document: queries.home.getRevenueData, variables: { period: 'day', date: 'last60' } } ]
);
```

Here's a [reference](https://redux-saga.js.org/docs/api#callcontext-fn-args) of call function from redux-saga documentation which mentions `args` should be in an array.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
